lib/ui/painting.dart

```diff
@@ -4451,6 +4433,103 @@ class Shadow {
   String toString() => 'TextShadow($color, $offset, $blurRadius)';
 }
 
+/// A handle to a read-only byte buffer that is managed by the engine.
+class ImmutableBuffer extends NativeFieldWrapperClass2 {
+  ImmutableBuffer._(this.length);
+
+  /// Creates a copy of the data from a [Uint8List] suitable for internal use
+  /// in the engine.
+  static Future<ImmutableBuffer> fromUint8List(Uint8List list) {
+    final ImmutableBuffer instance = ImmutableBuffer._(list.length);
+    return _futurize((_Callback<void> callback) {
+      instance._init(list, callback);
+    }).then((_) => instance);
+  }
+  void _init(Uint8List list, _Callback<void> callback) native 'ImmutableBuffer_init';
+
+  /// The length, in bytes, of the underlying data.
+  final int length;
+
+  /// Retreives a single byte from the underlying data.
+  int operator [](int index) => elementAt(index);
+
+  /// Retreives a single byte from the underlying data.
+  int elementAt(int index) {
+    if (index >= length || index < 0) {
+      throw RangeError.index(index, this);
+    }
+    return _elementAt(index);
+  }
+  int _elementAt(int index) native 'ImmutableBuffer_elementAt';
```

Comment:
A runtime call per byte access will be very inefficient. If this were staying here, I'd recommend caching a Uint8List in a field of this object that is backed by an external typed data.

Reply:
Per the comment thread above, this is being removed completely as there is no use case for it.